uiowa/uiowa
# Feature: Banner block refactor

Initial work to resolve https://github.com/uiowa/uiowa/issues/1928. 

# How to test banner block

```
blt ds --site=uiowa.edu
rm -rf ./node_modules
yarn cache clean
yarn install
yarn workspace uids_base gulp --development
drush @home.local cr
```

**Basic functionality**

1. Create a new banner block without using any of the "override options"
2. Create a banner block without a link
3. Crate a banner block without link text, is the banner still clickable?
4. Use any number of combinations in the "override options" area, do they work as expected?
5. Change the "Background" option to a pattern or background color, does it work different combinations in the "override options" area?

**Usage without layouts**

6. Move banner block around from one, two, three, and four column layouts.  Heading sizes and background opacity should change if used within two, three, and four column layouts. 

**Responsive behavior**

7. Collapse banners in different layout columns to different viewport sizes and note where it breaks.   Do we want to set a character count on the `Excerpt` to limit mobile overflow issues?

## Testing update hooks

1. Sync a site with existing, or add a hero and banner block to a site.
2. Run update hooks (8033, 8034).
3. Check that existing banner blocks still exist, now overriding styles with those marked `(default)`.
4. Check that, in place of hero blocks, banner blocks are present with the same content.
5. Check that any existing link in the original hero is present and functional as a banner button link.
6. Check that the newly created banner blocks are editable, and that the attached image can be replaced.
